plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakePartitionsTable.java

```diff
@@ -0,0 +1,235 @@
+import static io.trino.plugin.deltalake.transactionlog.DeltaLakeSchemaSupport.extractSchema;
+import static io.trino.plugin.deltalake.transactionlog.DeltaLakeSchemaSupport.getColumnMappingMode;
+import static io.trino.spi.block.RowValueBuilder.buildRowValue;
+import static io.trino.spi.type.BigintType.BIGINT;
+import static io.trino.spi.type.TypeUtils.writeNativeValue;
+import static java.util.Objects.requireNonNull;
+
+public class DeltaLakePartitionsTable
+        implements SystemTable
+{
+    private final SchemaTableName tableName;
+    private final String tableLocation;
+    private final TransactionLogAccess transactionLogAccess;
+    private final TypeManager typeManager;
+    private final ConnectorTableMetadata tableMetadata;
+    private final List<DeltaLakeColumnHandle> partitionColumns;
+    private final List<RowType.Field> partitionFields;
+
+    public DeltaLakePartitionsTable(
+            ConnectorSession session,
+            SchemaTableName tableName,
+            String tableLocation,
+            TransactionLogAccess transactionLogAccess,
+            TypeManager typeManager)
+    {
+        this.tableName = requireNonNull(tableName, "tableName is null");
+        this.tableLocation = requireNonNull(tableLocation, "tableLocation is null");
+        this.transactionLogAccess = requireNonNull(transactionLogAccess, "transactionLogAccess is null");
+        this.typeManager = requireNonNull(typeManager, "typeManager is null");
+
+        this.partitionColumns = getPartitionColumns(session, this.tableName, this.tableLocation);
+
+        this.partitionFields = this.partitionColumns.stream()
+                .map(column -> RowType.field(column.getBaseColumnName(), column.getType()))
+                .collect(toImmutableList());
+
+        ImmutableList.Builder<ColumnMetadata> columnMetadataBuilder = ImmutableList.builder();
+
+        if (!this.partitionFields.isEmpty()) {
+            columnMetadataBuilder.add(new ColumnMetadata("partition", RowType.from(this.partitionFields)));
+        }
+
+        columnMetadataBuilder.add(new ColumnMetadata("file_count", BIGINT));
+        columnMetadataBuilder.add(new ColumnMetadata("total_size", BIGINT));
+
+        this.tableMetadata = new ConnectorTableMetadata(tableName, columnMetadataBuilder.build());
+    }
+
+    @Override
+    public Distribution getDistribution()
+    {
+        return Distribution.SINGLE_COORDINATOR;
+    }
+
+    @Override
+    public ConnectorTableMetadata getTableMetadata()
+    {
+        return tableMetadata;
+    }
+
+    @Override
+    public ConnectorPageSource pageSource(ConnectorTransactionHandle transactionHandle, ConnectorSession session,
+                                          TupleDomain<Integer> constraint)
+    {
+        TableSnapshot tableSnapshot;
+
+        try {
+            // Verify the transaction log is readable
+            tableSnapshot = transactionLogAccess.loadSnapshot(session, tableName, tableLocation);
```

Comment:
We already load the snapshot in `getPartitionColumns` method. Please avoid redundant call.